resource/config.go

```diff
@@ -328,10 +328,10 @@ func (conf *Config) validate(path, defaultAPIType string) ([]string, error) {
 	return deps, nil
 }
 
-// A ConfigValidator validates a configuration and also
-// returns dependencies that were implicitly discovered.
+// A ConfigValidator validates a configuration and also returns both required and optional
+// dependencies that were implicitly discovered.
 type ConfigValidator interface {
-	Validate(path string) ([]string, error)
+	Validate(path string) (requiredDependencies, optionalDependencies []string, err error)
```

Comment:
This is the function signature change that forced all the other changes in this second commit. Used named returns here for clarity of order of parameters for future implementers.